src/Common/ColumnsHashing.h

```diff
@@ -83,8 +85,11 @@ struct HashMethodString
 
     HashMethodString(const ColumnRawPtrs & key_columns, const Sizes & /*key_sizes*/, const HashMethodContextPtr &)
     {
-        const IColumn & column = *key_columns[0];
-        const ColumnString & column_string = assert_cast<const ColumnString &>(column);
+        const IColumn * column = key_columns[0];
+        if (isColumnConst(*column))
```

Comment:
BTW, what about other modifiers like `Nullable`?

Reply:
That works, nullable columns use specific hashing method for integer and generic method for other types.
Single const string is a special case that uses String hash method but it didn't handle the const, and that was the case reported in the issue.